backend/src/main.rs

```diff
@@ -172,8 +180,89 @@ fn get_balance() -> Amount {
 }
 
 #[update]
-fn answer_task(_id: TaskId, _content: Content) -> AnswerId {
-    0
+fn answer_task(task_id: TaskId, content: Content) -> AnswerId {
+    let caller = caller();
+    STATE.with(|s| {
+        let ledger = s.ledger.borrow();
+
+        // Precondition: caller is a principal on the ledger
+        if !ledger.contains_key(&caller) {
+            ic_cdk::trap(&format!("Principal {} cannot provide an answer as this is not a registered\
+             user on the ledger.", caller));
+        }
+
+    });
+
+
+    STATE.with(|s| {
+        let mut tasks = s.tasks.borrow_mut();
+        let mut answers = s.answers.borrow_mut();
+
+        if let Some(task) = tasks.get_mut(&task_id) {
+
+            // Precondition: there are less than max_answers for taskID
+            if task.answers.len() > MAX_NUMBER_ANSWERS {
```

Comment:
```suggestion
            if task.answers.len() >= MAX_NUMBER_ANSWERS {
```
otherwise we allow going one above `MAX_NUMBER_ANSWERS`.